### PR TITLE
feat: support jammy

### PR DESCRIPTION
**Pull request by wep21**
To support humble/rolling distribution, I fixed CMakeLists.txt and updated ros2 ci.

### CODE REVIEW

File: .github/workflows/ros2-ci.yaml [hunk 1]
```diff
@@ -9,37 +9,44 @@ on:
 jobs:
 
   build-ros2:
-    runs-on: ubuntu-latest
+    runs-on: [ubuntu-latest]
```
Comment (kenji-miyake) [inline .github/workflows/ros2-ci.yaml]: To minimize the diff.

```suggestion
    runs-on: ubuntu-latest
```

File: .github/dependabot.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+  - package-ecosystem: github-actions
+    directory: /
+    schedule:
+      interval: daily
+    open-pull-requests-limit: 1
```
Comment (kenji-miyake) [inline .github/dependabot.yaml]: I think you can drop this from this PR.

File: .github/workflows/ros2-ci.yaml [hunk 3]
```diff
@@ -9,37 +9,44 @@ on:
...
+          - galactic
+          - humble
+          - rolling
         include:
-          - ros-version: foxy
-            image: ros:foxy
-          - ros-version: galactic
-            image: ros:galactic
+          # Foxy Fitzroy (June 2020 - May 2023)
+          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-foxy-ros-base-latest
```
Comment (kenji-miyake) [inline .github/workflows/ros2-ci.yaml]: Why did you change the containers from ROS's official ones?

File: .github/workflows/ros2-ci.yaml [hunk 4]
```diff
@@ -9,37 +9,44 @@ on:
...
+            ros_distribution: galactic
+            ros_version: 2
+          # Humble Hawksbill (May 2022 - May 2027)
+          - docker_image: rostooling/setup-ros-docker:ubuntu-jammy-ros-humble-ros-base-latest
+            ros_distribution: humble
+            ros_version: 2
+          # Rolling Ridley  (June 2020 - Present)
+          - docker_image: rostooling/setup-ros-docker:ubuntu-jammy-ros-rolling-ros-base-latest
+            ros_distribution: rolling
+            ros_version: 2
```
Comment (kenji-miyake) [inline .github/workflows/ros2-ci.yaml]: I guess we can drop the `ros_version` data.

```suggestion
          # Galactic Geochelone (May 2021 - November 2022)
          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-galactic-ros-base-latest
            ros_distribution: galactic
          # Humble Hawksbill (May 2022 - May 2027)
          - docker_image: rostooling/setup-ros-docker:ubuntu-jammy-ros-humble-ros-base-latest
            ros_distribution: humble
          # Rolling Ridley  (June 2020 - Present)
          - docker_image: rostooling/setup-ros-docker:ubuntu-jammy-ros-rolling-ros-base-latest
            ros_distribution: rolling
```

File: .github/workflows/ros2-ci.yaml [hunk 5]
```diff
@@ -9,37 +9,44 @@ on:
...
+          - humble
+          - rolling
         include:
-          - ros-version: foxy
-            image: ros:foxy
-          - ros-version: galactic
-            image: ros:galactic
+          # Foxy Fitzroy (June 2020 - May 2023)
+          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-foxy-ros-base-latest
+            ros_distribution: foxy
```
Comment (kenji-miyake) [inline .github/workflows/ros2-ci.yaml]: [nits/IMO] I feel writing the key data first is better.

```suggestion
          - ros_distribution: foxy
            docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-foxy-ros-base-latest
```

File: .github/workflows/ros2-ci.yaml [hunk 6]
```diff
@@ -9,37 +9,44 @@ on:
...
-          colcon test --event-handlers console_cohesion+; \
-          colcon test-result --verbose'
+      - name: setup directories
+        run: mkdir -p ros_ws/src
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          path: ros_ws/src
+      - name: build and test
+        uses: ros-tooling/action-ros-ci@v0.2
```
Comment (kenji-miyake) [inline .github/workflows/ros2-ci.yaml]: Why did you change the CI scripts drastically?
If there is no specific reason, generally, it's recommended to keep the diff minimum.

File: CMakeLists.txt [hunk 7]
```diff
@@ -66,7 +66,7 @@ ExternalProject_Add(tvm-build
 ExternalProject_Add_Step(tvm-build dmlc-build
   DEPENDEES patch
   WORKING_DIRECTORY <SOURCE_DIR>/3rdparty/dmlc-core
-  COMMAND ${CMAKE_COMMAND} --configure -DCMAKE_POSITION_INDEPENDENT_CODE=ON .
+  COMMAND ${CMAKE_COMMAND} -DCMAKE_POSITION_INDEPENDENT_CODE=ON .
```
Comment (kenji-miyake) [inline CMakeLists.txt:69]: For what did you have to remove the `--configure`?

**Review by kenji-miyake — COMMENTED** [review]
Actually, I'm not a maintainer of this repository, but I added general comments.

**Comment by kenji-miyake** [comment]
@LucaFos Could you resolve my comments? It seems I have no rights. :cry: 
Ah, probably @wep21 has the right and resolved. :pray:

File: .github/workflows/ros2-ci.yaml (hunk 6 above)
Comment (wep21) [reply .github/workflows/ros2-ci.yaml]: @kenji-miyake 
I adopted your suggestion at https://github.com/autowarefoundation/tvm_vendor/pull/9/commits/f36f3f4ce3a5015f09dae20cf81ac2a48512c533.

File: CMakeLists.txt (hunk 7 above)
Comment (wep21) [reply CMakeLists.txt:69]: It has caused the following error.
```
CMake Error: Unknown argument --configure
```
My cmake version is
```
❯ cmake --version
cmake version 3.22.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

**Review by ambroise-arm — APPROVED** [review]
looks good to me
